lodash.js

```diff
@@ -11405,6 +11424,37 @@
     }
 
     /**
+     * Updates the value at `path` of `object` with given `func`. If a portion of `path` doesn't exist
+     * it's created. Arrays are created for missing index properties while objects
+     * are created for all other missing properties.
+     *
+     * @static
+     * @memberOf _
+     * @category Object
+     * @param {Object} object The object to modify.
+     * @param {Array|string} path The path of the property to set.
+     * @param {Function} func The updater function.
+     * @returns {Object} Returns `object`.
+     * @example
+     *
+     * var object = { 'a': [{ 'b': { 'c': 3 } }] };
+     *
+     * _.update(object, 'a[0].b.c', function (n) { return n + 10; });
+     * console.log(object.a[0].b.c);
+     * // => 13
+     *
+     * _.update(object, 'x[0].y.z', function (n) { return n ? n + 1 : 0; });
+     * console.log(object.x[0].y.z);
+     * // => 0
+     */
+    function update() {
+      var args = Array.prototype.slice.call(arguments),
+          funcargs = args.splice(3);
+      args.push(funcargs);
+      return baseUpdate.apply(null, args);
+    }
+
```

Comment:
Lets simplify this a bit and remove the partial args stuff as that can be done with `_.partialRight` and friends.